plugins/modules/system/disk_info.py

```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/python
+#
+# Copyright 2021 VMware, Inc
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+
+DOCUMENTATION = '''
+---
+module: disk_info
+version_added: "2.5.0"
+short_description: Retrieves disk info in the Linux/Unix systems
+description:
+    - This module retrieves disk info like free space, used percentage, mount location, and so on.
+requirements:
+  - psutil (Python module)
+options:
+    name:
+        description:
+            - This is the disk name to query, if this parameter is not passed, it queries all the disk information.
+        default: all
+        type: str
+    filter:
+        description:
+            - This is the parameter used to filter only the specific information from the output like freespace, mountpoint, fstype, and so on.
+        type: list
+        elements: str
+author:
+    - Saranya Sridharan (@saranyasridharan)
+    - Swetha M (@swetm)
+    - Nagendra Kini (@NagendraKini)
+    - Anup Kumar (@anuppoojari)
+    - Mehmoodkha Pathan (@Mehmoodpathan)
+'''
+
+EXAMPLES = '''
+- name: Get disk info
+  community.general.disk_info:
+
+- name: Get Disk info for Particular disk info
+  community.general.disk_info:
+    name: diskname
+
+- name: Filter the output parameters for specific disk
+  community.general.disk_info:
+    name: diskname
+    filter:
+      - mountpoint
+      - freespace
+      - totalsize
+
+- name: Filter the output parameters for all disk
+  community.general.disk_info:
+    filter:
+      - fstype
+      - usedspace
+      - capacity_percentage
+'''
+
+
+RETURN = '''
+ansible_facts:
+    description: disk information only filtered parameter values if filter option is given.
+    returned: dictionary of disks informations
+    type: dict
+    sample: {
+        '/dev/sda': {
+             'capacity_percentage': '2%'
+        }
+    }
+'''
+
+
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+import traceback
+PSUTIL_IMP_ERR = None
+try:
+    import psutil
+    HAS_PSUTIL = True
+except ImportError:
+    PSUTIL_IMP_ERR = traceback.format_exc()
+    HAS_PSUTIL = False
+
+
+# To get the disk partitions information
+def disk_info(devicename):
+    output = {}
+    partitions = psutil.disk_partitions()
+    found = False
+    for partition in partitions:
+        if not partition.device.startswith("/dev/"):
```

Comment:
Why this line?

Reply:
I have a task like this in my test to get only the physical disk information , "df -h | grep '^/dev/' | awk '{ print $1 }'" to filter the disk details like tmpfs. Its disk count was mismatching with the output of this module , I was seeing 'shm' disk information also. So I thought to add this line to handle it.

Reply:
I would really remove it. There's no reason to limit to devices in `/dev/`, as in theory devices can also be somethere else in the FS tree. (How often that happens in practice I cannot tell...)